src/firefly/js/visualize/ui/CatalogSelectViewPanel.jsx

```diff
@@ -0,0 +1,623 @@
+}
+function hideSearchPanel() {
+    dispatchHideDropDown();
+}
+
+function doCatalog(request) {
+
+    const conesize = convertAngle('deg', 'arcsec', request.conesize);
+    var title = request.project + '-' + request.cattable + ' (' + request.spatial;
+    if (request.spatial === SpatialMethod.Box.value || request.spatial === SpatialMethod.Cone.value || request.spatial === SpatialMethod.Elliptical.value) {
+        title += ':' + conesize;
+    }
+    title += ')';
+    var tReq = TableRequest.newInstance({
+
+        id: 'GatorQuery',
+        title: title,
+        SearchMethod: request.spatial,
+        catalog: request.cattable,
+        RequestedDataSet: request.catalog,
+        use: 'catalog_overlay',
+        catalogProject: request.project
+
+    });
+
+    // change and merge others parameters in request if elliptical
+    // plus change spatial name to cone
+    // (Gator search method for elliptical is cone)
+    if (request.spatial === SpatialMethod.Elliptical.value) {
+
+        const pa = get(request, 'posangle', 0);
+        const ar = get(request, 'axialratio', 0.26);
+
+        // see PA and RATIO string values in edu.caltech.ipac.firefly.server.catquery.GatorQuery
+        merge(tReq, {'posang': pa, 'ratio': ar});
+    }
+
+    if (request.spatial === SpatialMethod.Cone.value
+        || request.spatial === SpatialMethod.Box.value
+        || request.spatial === SpatialMethod.Elliptical.value) {
+        merge(tReq, {[ServerParams.USER_TARGET_WORLD_PT]: request[ServerParams.USER_TARGET_WORLD_PT]});
+        if (request.spatial === SpatialMethod.Box.value) {
+            merge(tReq, {'size': conesize});
+        }else{
+            merge(tReq, {'radius': conesize});
+        }
+    }
+
+    if (request.spatial === SpatialMethod.Polygon.value) {
+        const p = escape(request.polygoncoords);//encodeURI
+        encodeURI(request.polygoncoords);
+        merge(tReq, {'polygon': p});
+    }
+
+    dispatchTableSearch(tReq);
+}
+
+function doVoSearch(request) {
+    var tReq = TableRequest.newInstance({
+        [ServerParams.USER_TARGET_WORLD_PT]: request[ServerParams.USER_TARGET_WORLD_PT],
+        id: 'GatorQuery',
+        title: request.catalog,
+        SearchMethod: request.spatial,
+        catalog: request.cattable,
+        RequestedDataSet: request.catalog,
+        radius: request.conesize,
+        use: 'catalog_overlay',
+        catalogProject: request.project
+    });
+    console.log('Does not dispatch yet ' + tReq);
+    //dispatchTableSearch(tReq);
+}
+
+function doLoadTable(request) {
+    var tReq = TableRequest.newInstance({
+        id: 'GatorQuery',
+        title: request.catalog,
+        SearchMethod: request.spatial,
+        catalog: request.cattable,
+        RequestedDataSet: request.catalog,
+        radius: request.conesize,
+        use: 'catalog_overlay',
+        catalogProject: request.project
+    });
+    console.log('Does not dispatch yet ' + tReq);
+    //dispatchTableSearch(tReq);
+}
```

Comment:
Just a heads up, I've removed TableRequest.newInstance in favor of custom creator functions, ie. makeIrsaCatalogRequest()
This is part of the API changes.